init/main.go

```diff
@@ -787,7 +787,30 @@ func readStartTime() {
 	}
 }
 
+func enableLocalEcho() error {
+	stdin := os.Stdin
+	fd := int(stdin.Fd())
+
+	termios, err := unix.IoctlGetTermios(fd, unix.TCGETS)
+	if err != nil {
+		return err
+	}
+
+	newState := *termios
+	newState.Lflag |= unix.ECHO
+	if err := unix.IoctlSetTermios(fd, unix.TCSETS, &newState); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func emergencyShell() {
+	// Force local echo (might have been disabled by readPassword).
+	if err := enableLocalEcho(); err != nil {
```

Comment:
let's move the `enableLocalEcho` call under `if _, err := os.Stat("/usr/bin/busybox"); !os.IsNotExist(err) {` so the echo will be disable only if it is needed.

Reply:
Good catch, fixed.